cpp/lib/src/JSON.cc

```diff
@@ -781,4 +783,23 @@ int64_t LookupInteger(const std::string &path, const JSONNode * const tree, cons
 }
 
 
+std::string EscapeString(const std::string unescaped_string) {
+    std::ostringstream escaped_string;
+    for (auto iter = unescaped_string.begin(); iter != unescaped_string.end(); iter++) {
```

Comment:
Use an implicit for loop, like this:  for (const char ch : unescaped_string) { ...